pkg/nodestats/nodestats.go

```diff
@@ -0,0 +1,76 @@
+package nodestats
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/iotexproject/iotex-core/blocksync"
+	"github.com/iotexproject/iotex-core/p2p"
+	"github.com/iotexproject/iotex-core/pkg/routine"
+)
+
+const (
+	// PeriodicReportInterval is the interval for generating periodic reports
+	PeriodicReportInterval = 5 * time.Minute
+)
+
+// NodeStats is the struct for getting node stats
+type NodeStats struct {
+	api       *APILocalStats
+	system    *systemStats
+	blocksync blocksync.BlockSync
+	p2pAgent  p2p.Agent
+	task      *routine.RecurringTask
+}
```

Comment:
need to move to a diff place (maybe `chainservice` or `server`?) since the struct included `blocksync` and `p2p`